### PR TITLE
Avoid fixed-name collisions

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

**Comment by mattmoor** [comment]
This manifests in net-contour as 404s:
```
    util.go:886: Error meeting response expectations: got unexpected status: 404, expected map[200:{}]
    util.go:886: HTTP/1.1 404 Not Found
        Date: Sun, 23 Aug 2020 16:18:06 GMT
        Server: envoy
        Content-Length: 0
```

This is because the conflicting routes are pulled from the programming sent to Envoy, and after it's reported ready this can lead to failures.  This can also lead to the sorts of hangs we've seen becoming ready if we hit this concurrently before either becomes ready.

File: test/conformance/ingress/visibility.go [hunk 1]
```diff
@@ -71,19 +71,20 @@ func TestVisibility(t *testing.T) {
...
 	}
 }
 
-func testProxyToHelloworld(t *testing.T, ingress *v1alpha1.Ingress, clients *test.Clients, name, privateHostName string) {
+func testProxyToHelloworld(t *testing.T, ingress *v1alpha1.Ingress, clients *test.Clients, privateHostName string) {
 
 	loadbalancerAddress := ingress.Status.PrivateLoadBalancer.Ingress[0].DomainInternal
 	proxyName, proxyPort, _ := CreateProxyService(t, clients, privateHostName, loadbalancerAddress)
 
 	// Using fixed hostnames can lead to conflicts when -count=N>1
```
Comment (tcnghia) [inline test/conformance/ingress/visibility.go:84]: @nak3 this makes me :rofl:

Comment (tcnghia) [reply test/conformance/ingress/visibility.go:84]: (about the left over comment)

**Comment by nak3** [comment]
/lgtm

Thank you.

**Pull request merged**